src/code/z_demo.c

```diff
@@ -1,99 +1,1650 @@
+        D_801BB128 = 0;
+        csCtx->playerAction = NULL;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EDA84.s")
+        for (i = 0; i < ARRAY_COUNT(csCtx->actorActions); i++) {
+            csCtx->actorActions[i] = NULL;
+        }
 
+        csCtx->state++;
+        if (csCtx->state == CS_STATE_1) {
+            Audio_SetCutsceneFlag(1);
+
+            csCtx->frames = 0xFFFF;
+            csCtx->csCamId = ActorCutscene_GetCurrentCamera(0x7F);
+            func_8016119C(Play_GetCamera(globalCtx, csCtx->csCamId), &sCutsceneCameraInfo);
+            csCtx->unk_18 = 0xFFFF;
+
+            if (gSaveContext.cutsceneTrigger == 0) {
+                Interface_ChangeAlpha(1);
+                ShrinkWindow_SetLetterboxTarget(0x20);
+                ShrinkWindow_SetLetterboxMagnitude(0x20);
+                csCtx->state++;
+            }
+
+            func_800ED980(globalCtx, csCtx);
+        }
+
+        gSaveContext.cutsceneTrigger = 0;
+    }
+}
+
+#ifdef NON_MATCHING
+// HandleFlags?
+void func_800EDBE0(GlobalContext* globalCtx) {
+    CutsceneEntry* temp_a3;
+    s16 sp2A;
+    SceneTableEntry* sp24;
+    s32 temp_v0_3;
+
+    if (((gSaveContext.gameMode == 0) || (gSaveContext.gameMode == 1)) && (gSaveContext.respawnFlag <= 0)) {
+        sp2A = func_800F21CC();
+        if (sp2A != -1) {
+            temp_v0_3 = func_800F2138(sp2A);
+            if (temp_v0_3 != -1) {
+                temp_a3 = ((void)0, globalCtx->csCtx.sceneCsList);
+                if ((temp_a3[temp_v0_3].unk7 != 0xFF) && (gSaveContext.respawnFlag == 0)) {
+                    if (temp_a3[temp_v0_3].unk7 == 0xFE) {
+                        ActorCutscene_Start(sp2A, NULL);
+                        gSaveContext.showTitleCard = false;
+                    } else if (!((1 << (temp_a3[temp_v0_3].unk7 % 8)) &
+                                 gSaveContext.weekEventReg[temp_a3[temp_v0_3].unk7 / 8])) {
+                        gSaveContext.weekEventReg[(temp_a3[temp_v0_3].unk7 / 8)] |= 1 << (temp_a3[temp_v0_3].unk7 % 8);
+                        ActorCutscene_Start(sp2A, NULL);
+                        gSaveContext.showTitleCard = false;
+                    }
+                }
+            } else {
+                ActorCutscene_StartAndSetUnkLinkFields(sp2A, NULL);
+            }
+        }
+    }
+
+    if ((gSaveContext.respawnFlag == 0) || (gSaveContext.respawnFlag == -2)) {
+        sp24 = globalCtx->loadedScene;
+        if ((sp24->titleTextId != 0) && gSaveContext.showTitleCard) {
+            if ((Entrance_GetTransitionFlags(gSaveContext.sceneSetupIndex + gSaveContext.entranceIndex) & 0x4000) !=
+                0) {
+                func_80151A68(globalCtx, sp24->titleTextId);
+            }
+        }
+
+        gSaveContext.showTitleCard = true;
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EDBE0.s")
+#endif
+
+void nop_800EDDB0(GlobalContext* globalCtx) {
+}
+
+void func_800EDDBC(UNK_TYPE arg0, UNK_TYPE arg1) {
+}
```

Comment:
```suggestion
void Cutscene_Unused1(GlobalContext* globalCtx) {
}

void Cutscene_Unused2(UNK_TYPE arg0, UNK_TYPE arg1) {
}
```
Just as an idea

Reply:
Unused1 is actually used by z_play

Reply:
Ah good to know. Then yeah, Unused is a bad name. Then maybe `Cutscene_Noop` and `Cutscene_Unused` for these two?

Reply:
I think we should wait for z_play before naming this function, it _could_ make a bit more of sense with more context.
Also, I'm not really convinced by naming functions "unused"

Reply:
Hmmm yeah alright that's fair on both points. I think I'd still prefer the intermediate `nop_800EDDB0` -> `func_800EDDB0`. But you can leave it if you'd like